packages/core/src/components/popover/popover.tsx

```diff
@@ -612,30 +616,27 @@ export class Popover<
         return popperModifiers;
     }
 
-    private handleTargetFocus = (e: React.FocusEvent<HTMLElement>) => {
+    private handleTargetFocus = (e: React.FocusEvent<HTMLElement, HTMLElement>) => {
         if (this.props.openOnTargetFocus && this.isHoverInteractionKind()) {
             if (e.relatedTarget == null && !this.lostFocusOnSamePage) {
                 // ignore this focus event -- the target was already focused but the page itself
                 // lost focus (e.g. due to switching tabs).
                 return;
             }
-            this.handleMouseEnter(e as unknown as React.MouseEvent<HTMLElement>);
+            this.handleMouseEnter(e);
         }
     };
 
-    private handleTargetBlur = (e: React.FocusEvent<HTMLElement>) => {
+    private handleTargetBlur = (e: React.FocusEvent<HTMLElement, HTMLElement>) => {
         if (this.props.openOnTargetFocus && this.isHoverInteractionKind()) {
             if (e.relatedTarget != null) {
-                // if the next element to receive focus is within the popover, we'll want to leave the
-                // popover open.
-                if (
-                    e.relatedTarget !== this.popoverElement &&
-                    !this.isElementInPopover(e.relatedTarget as HTMLElement)
```

Comment:
This is the main fix in this PR. `Overlay2` brings focus to its `startFocusTrapElement` upon `autoFocus`, and this element is **not** within the popover element. It is just outside the popover element, within the `overlay` element. So, use `isElementInOverlay` instead of `isElementInPopover`